src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/BitbucketWebhookHandler.java

```diff
@@ -106,157 +104,94 @@ private void deleteWebhooks(List<BitbucketWebhook> webhooks) {
     }
 
     /**
-     * Returns the correct webhook event to subscribe to.
+     * Returns the webhook events to subscribe to.
      * For Mirror sync event, the input request should point to mirror.
-     * 1. If the webhook capability does not contain mirror sync, this version with webhooks at least supports ref change
-     * For Repo ref event,
-     * 1. Input request does not point to mirrors
-     * 2. Since this version supports webhooks, it will definitely support pull requests and ref changes
-     * 3. We subscribe events depending on if we want to only listen for ref changes, only pull request events or both
-     * If we subscribe to pull request events, we subscribe to all of these: when pull requests are opened, closed, deleted or merged.
+     * If the webhook capability does not contain mirror sync, this version with webhooks at least supports ref change,
+     * since this version supports webhooks, it will definitely support pull requests and ref changes
+     * We subscribe to events depending on if we want to only listen for ref changes, only pull request events or both
+     * If we subscribe to pull request events, we subscribe to all of these: pull requests are opened,
+     * closed, deleted or merged.
      *
      * @param request the input request
      * @return the correct webhook event
```

Comment:
I'm not sure this is correct (we return a collection of events, and I'm not sure what is meant by "correct" here- perhaps matching?)